lang/cs/Org.Apache.REEF.Bridge.Core.Tests/Fail/Driver/NoopTask.cs

```diff
@@ -0,0 +1,149 @@
+using Org.Apache.REEF.Utilities;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Bridge.Core.Tests.Fail.Driver
+{
+    internal sealed class NoopTask : 
+        ITask, 
+        ITaskMessageSource, 
+        IDriverMessageHandler, 
+        IObserver<ISuspendEvent>, 
+        IObserver<ITaskStop>, 
+        IObserver<ICloseEvent>
+    {
+        private static readonly Logger Log = Logger.GetLogger(typeof(NoopTask));
+
+        private static readonly TaskMessage InitMessage =
+            TaskMessage.From("nooptask", ByteUtilities.StringToByteArrays("MESSAGE::INIT"));
+
+        private bool _isRunning = true;
+        private Optional<TaskMessage> _message = Utilities.Optional<TaskMessage>.Empty();
+
+        [Inject]
+        private NoopTask()
+        {
+            Log.Log(Level.Info, "NoopTask created.");
+        }
+
+        public byte[] Call(byte[] memento)
+        {
+            _isRunning = true;
+            while (_isRunning)
+            {
+                try
+                {
+                    Log.Log(Level.Info, "NoopTask.call(): Waiting for the message.");
+                    lock (this)
+                    {
+                        Monitor.Wait(this);
+                    }
+                }
+                catch (System.Exception ex)
+                {
+                    Log.Log(Level.Warning, "NoopTask.wait() interrupted.", ex);
+                }
+            }
+
+            Log.Log(Level.Info,
+                "NoopTask.call(): Exiting with message {0}",
+                ByteUtilities.ByteArraysToString(_message.OrElse(InitMessage).Message));
+            return _message.OrElse(InitMessage).Message;
+        }
+
+
+
+        public Optional<TaskMessage> Message
+        {
+            get
+            {
+                Log.Log(Level.Info,
+                    "NoopTask.getMessage() invoked: {0}",
+                    ByteUtilities.ByteArraysToString(_message.OrElse(InitMessage).Message));
+                return _message;
+            }
+        }
+
+        public void Dispose()
+        {
+            Log.Log(Level.Info, "NoopTask.stopTask() invoked.");
+            _isRunning = false;
+            lock (this)
+            {
+                Monitor.Pulse(this);
+            }
+        }
+
+        public void OnError(System.Exception error)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void OnCompleted()
+        {
+            throw new NotImplementedException();
+        }
+
+        /**
+         * Handler for SuspendEvent.
+         */
+        public void OnNext(ISuspendEvent suspendEvent)
+        {
+            Log.Log(Level.Info, "NoopTask.TaskSuspendHandler.OnNext() invoked.");
+            Dispose();
+        }
+
+        /**
+         * Handler for TaskStop.
+         */
+        public void OnNext(ITaskStop value)
+        {
+            Log.Log(Level.Info, "NoopTask.TaskStopHandler.OnNext() invoked.");
```

Comment:
Should Dispose be called here?

Reply:
The REEF Evaluator calls Dispose on our behalf.

Reply:
Personally I don't like that ITask : IDisposable because of such confusion. @markusweimer FYI